velero-plugins/pod/restore.go

```diff
@@ -50,8 +50,19 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	if err != nil {
 		return nil, err
 	}
-	// Check if pod has owner Refs
-	if len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" {
+
+	// get backup associated with the restore
+	backupName := input.Restore.Spec.BackupName
+	backup, err := common.GetBackup(backupName, input.Restore.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// check for default restic flag
+	defaultVolumesToResticFlag := backup.Spec.DefaultVolumesToRestic
+
+	// Check if pod has owner Refs and defaultVolumesToRestic flag as false/nil
+	if len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" && (defaultVolumesToResticFlag == nil || !*defaultVolumesToResticFlag) {
```

Comment:
We may want to warn the user if `defaultVolumesToResticFlag` is set to `nil` that we are skipping the restore of the pod and it may break things if this was a server config they expected to see reflected at the backup time without setting it on the backup CR

Reply:
@dymurray I don't think that's necessary, since it's basically impossible for it to be nil if the backup has run already. It's probably safe to consider it false.

Reply:
Ignore... Scott brought it to my attention that it should really never be nil in this situation

Reply:
As discussed this never will be nil because of the backup controller